### ISSUE TITLE
AppExtension incompatibility

### ISSUE THREAD

**Issue by Gozala** [textileio/ios-textile#44]
I am unable to use library in the AppExtension context as lifecycle manager assumes `UIApplication.sharedApplication` which does not exists:

```
/Users/gozala/Projects/textile-files/Pods/Textile/Textile/Classes/LifecycleManager.m:38:46: 'sharedApplication' is unavailable: not available on iOS (App Extension) - Use view controller based solutions where appropriate instead.
```

**Comment by asutula** [comment]
Ah interesting. I need to read up more about app extensions. There are a couple possibilities here.

1. Run all Textile node related code in the containing app (not the app extension) and the extension just accesses data somehow. This is better because the node can be launched periodically to sync data with it's peers (because the user opened the app or by using background tasks etc)
2. Figure out if it's actually appropriate to run Textile inside of an app extension. This seems unlikely because the node would only run when the extension is launched (which won't work well, the node is rarely online).

**Comment by asutula** [comment]
I will read more about extensions, but for those reasons, I'd suggest looking at running the node from the containing app.

**Comment by Gozala** [comment]
Here is my example app https://github.com/Gozala/ios-textile-drive uncommenting following file wild reproduce build failure https://github.com/Gozala/ios-textile-drive/blob/master/Podfile#L9

**Comment by Gozala** [comment]
> I will read more about extensions, but for those reasons, I'd suggest looking at running the node from the containing app.

I don't believe extension has a way of calling into a containing app, so it would not really work

**Comment by Gozala** [comment]
It seems [there is a way to do IPC](http://ddeville.me/2015/02/interprocess-communication-on-ios-with-mach-messages), however comment on the post suggests app will only be able to respond if it's in the foreground, which seems unlikely given that an extension will be used

> Thanks for the detail inputs. I am using with Appgroup on iOS 8 with 2 applications communicating each other. The server was not able to receive the message when it was in background mode. When the server app, comes to foreground, all the messages sent by the client are received. To verify I have enabled all the background and tested too.

**Comment by Gozala** [comment]
Another interesting find on the IPC https://github.com/mutualmobile/MMWormhole

**Comment by asutula** [comment]
Very interesting. Seems worth exploring.

**Comment by Gozala** [comment]
I had this thought that I think worth considering:

MMWormhole seems to rely on the fact that extension and app have a shared App Group giving them access to the set of files that are essentially used for messaging. Maybe Textile lib could live in both extension and an app context, assuming they read / write into same files, it probably would not matter if extension lifetime is short because once it's activated it can boot node from the files that app had being writing into & there for server the purpose. As of publishing / writing new content extension can just add stuff to the local IPFS node, and let the app process do the networking. 

@asutula What I do not know how visible it is in practice to boot node with changed repo & expect it to pick up all the changes. Also given both processes are active at the same time there could be races, but I imagine with immutable nature of IPFS surface is probably limited so maybe still visible ?
